src/createUseQuery.ts

```diff
@@ -21,7 +23,7 @@ export function createUseQuery(createQuery: CreateQuery, action: QueryAction, pa
 
     query.dispose()
 
-    if(parameters === null) {
+    if(parameters === null || !immediate) {
```

Comment:
Yeah I don't think this is enough. We need to wrap the execute method possibly and tract that we no longer need to bail out.

Reply:
Ah yeah you're right. I have some ideas I'll explore

Reply:
one side effect of this decision is that if there are 2 queries, 1 with immediate true and the other false, both will operate as though immediate was false. I added tests which prove this to be the case. 

We _could_ give the user the perception that `execute` has not happened yet by just hard coding `executed: false` and `response: undefined`, etc.

Reply:
Is there something you found that made this not work well in the composition?

Reply:
I don't think this approach will work because a compositions query group can change. And immediate should only effect the first execution. Once execute has been called, that group and any future groups should execute as normal. I believe as is if the group changes the user will also need to call execute again.

Reply:
I think I solved for that in the createUseQuery service

Reply:
Still wondering about this

> Is there something you found that made this not work well in the composition?

I imagined this as a very simple change to useQuery. Since the composition has to be called during setup, `immediate` has a strong use case there. Where it doesn't in the query function. So exposing this globally is odd to me and more complex.